### PR TITLE
Pattern syntax docs touchup

### PR DESCRIPTION
**Pull request by mschwager**
Gave the pattern syntax a once-over and noticed a few places that could be improved/updated to the latest information.

### CODE REVIEW

File: docs/writing-rules/pattern-syntax.md [hunk 1]
```diff
@@ -316,65 +329,15 @@ baz.qux()
...
-```go
-crypto.hash("MD5", "text")
-
-const algorithm = "MD5"
-crypto.hash(algorithm, "text")
-
-```
-
-Java:
+Semgrep performs basic constant propagation.
```
Comment (mschwager) [inline docs/writing-rules/pattern-syntax.md:332]: This functionality can now be assumed true for all GA languages, this wasn't the case when this documentation was originally written.

**Comment by mschwager** [comment]
CI failure looks unrelated to these changes: https://rubydoc.info/docs/rails/ActiveSupport/JSON/Encoding.escape_html_entities_in_json= is giving a 521

File: docs/writing-rules/pattern-syntax.md [hunk 2]
```diff
@@ -139,14 +139,28 @@ The ellipsis operator can match any number of arguments to binary operations. Th
 foo = 1 + 2 + 3 + 4
 ```
 
-## Arrays
+## Containers
```
Comment (pabloest) [inline docs/writing-rules/pattern-syntax.md:142]: Note if there’s anything pointing to the previous header at `pattern-syntax/#arrays` (such as a CLI link) then you’ll want to add a manual html anchor link with the `#arrays` name.

File: docs/writing-rules/pattern-syntax.md [hunk 3]
```diff
@@ -190,7 +204,7 @@ if can_make_request:
 
 # Metavariables
 
-Metavariables are an abstraction to match code when don’t know the value or contents ahead of time, acting like [capture groups](https://regexone.com/lesson/capturing_groups) in regular expressions.
+Metavariables are an abstraction to match code when you don’t know the value or contents ahead of time. Like [capture groups](https://regexone.com/lesson/capturing_groups) in regular expressions.
```
Comment (pabloest) [inline docs/writing-rules/pattern-syntax.md:207]:
```suggestion
Metavariables are an abstraction to match code when you don’t know the value or contents ahead of time, similar to [capture groups](https://regexone.com/lesson/capturing_groups) in regular expressions.
```

**Review by pabloest — APPROVED** [review]

File: docs/writing-rules/pattern-syntax.md (hunk 2 above)
Comment (mschwager) [reply docs/writing-rules/pattern-syntax.md:142]: Nothing that I'm aware of!

**Pull request merged**